extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -356,6 +356,38 @@ export class Bind {
     return this.setStatePromise_;
   }
 
+  /**
+   * Merges a state object into the current global state, or parses a state
+   * expression and merges it into the global state.
+   * @param {(!JsonObject|string)} state
+   * @return {!Promise}
+   */
+  setStateAndUpdateHistory(state) {
+    // Support expressions
+    if (typeof state === 'string') {
+      // Emulate UIEvent 'click'
+      return this.pushStateWithExpression(
```

Comment:
A bit confusing if this API can both "set" and "push" state. I think it's worth separating into two APIs. 

Though for the sake of keeping this PR small, can we remove this case?

Reply:
To make sure I understand, are you suggesting that we do not update history with the `shadowDoc.setState()` method?

Reply:
I mean removing this case where passing a string calls "push" instead of "set". The former appends a new entry onto the history stack.

Reply:
Or maybe you just meant to call `setStateWithExpression` here?